### PR TITLE
mobile multiplatform plugin

### PR DESCRIPTION
**Pull request by Alex009**
**Summary:**

- Configured KMM module to iOS integration by [mobile-multiplatform-gradle-plugin](https://github.com/icerockdev/mobile-multiplatform-gradle-plugin);
- Fixed iOS compilation errors;
- AppAuth connected by CocoaPods;
- moko-mvvm swift additions connected by CocoaPods;
- kotlin 1.4.21;
- coroutines 1.4.2-native-mt;
- configured [export](https://kotlinlang.org/docs/reference/mpp-build-native-binaries.html#export-dependencies-to-binaries) of moko-mvvm and moko-resources to iOS header;
- fixed `build` task execution.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -18,4 +18,7 @@
 
 kotlin.code.style=official
 org.gradle.daemon=true
-org.gradle.jvmargs=-Xmx2560m
\ No newline at end of file
+org.gradle.jvmargs=-Xmx2560m
+
+kotlin.native.enableDependencyPropagation=false
```
Comment (f8full) [inline gradle.properties:23]: I'm quite curious about those two lines ^^

Comment (Alex009) [reply gradle.properties:23]: it's for enable iosMain support with commonizer. 
https://kotlinlang.org/docs/reference/migrating-multiplatform-project-to-14.html#try-the-hierarchical-project-structure

Comment (f8full) [reply gradle.properties:23]: Thks!

**Comment by f8full** [comment]
Thank you so so much @Alex009 for working on this!

Here is a description of the steps I took to test the PR

- install cocoapods
`pod install` failed because no Java
- install Java from java.com
`pod install`success
- run command failed because Gradle Java path was pointing to what looked like a `homebrew` install
- install `homebrew`
- install `openjdk8`
`pod install` failed again, complaining about X Code setup
- reinstall x code
- reinstall x code command line tools
`pod install` still complaining
- follow https://apple.stackexchange.com/a/336589 to setup command line tools in Xcode
- run configuration fails `please specify x code project in gradle.properties` which is done. Could not fix this issue that seems to have to do with the KMM plugin unable to grab the x code project scheme (I'm not even super sure what that means)

But, it works from X code!!
image

Once again, THANK YOU!

P.S.: I am not sure what to do with the resulting changelist 😅 , specifically this. I will commit and see :)
image

**Pull request merged**